build-scripts/config_common.cmake

```diff
@@ -52,23 +52,25 @@ if (CMAKE_BUILD_TYPE STREQUAL "Debug")
   add_definitions(-DBH_DEBUG=1)
 endif ()
 
-if (CMAKE_SIZEOF_VOID_P EQUAL 8)
-  if (WAMR_BUILD_TARGET STREQUAL "X86_64" OR WAMR_BUILD_TARGET STREQUAL "AMD_64"
-      OR WAMR_BUILD_TARGET MATCHES "AARCH64.*" OR WAMR_BUILD_TARGET MATCHES "RISCV64.*")
-    if (NOT WAMR_BUILD_PLATFORM STREQUAL "windows")
-      # Add -fPIC flag if build as 64-bit
+if (WAMR_BUILD_TARGET STREQUAL "X86_64" OR WAMR_BUILD_TARGET STREQUAL "AMD_64"
+    OR WAMR_BUILD_TARGET MATCHES "AARCH64.*" OR WAMR_BUILD_TARGET MATCHES "RISCV64.*")
+  if (NOT WAMR_BUILD_PLATFORM STREQUAL "windows")
+    # Add -fPIC flag if build as 64-bit
+    if (CMAKE_SIZEOF_VOID_P EQUAL 8)
       set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
       set (CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_C_FLAGS} -fPIC")
     endif ()
-  else ()
+  endif ()
+else ()
+  if (CMAKE_SIZEOF_VOID_P EQUAL 4)
```

Comment:
The change is invalid: when host is 64-bit (CAMKE_SIZEOF_VOID_P is 8) and target is set x86-32, it won't be handled, and flag `-m32` isn't passed to GCC, which causes compilation error, see CI: https://github.com/bytecodealliance/wasm-micro-runtime/runs/6964367552?check_suite_focus=true#step:6:138

How about not changing the handling of case `if (CMAKE_SIZEOF_VOID_P EQUAL 8)`, and adding the handling of case `(CMAKE_SIZEOF_VOID_P EQUAL 4)`:
```cmake
if (CMAKE_SIZEOF_VOID_P EQUAL 8)
   keep unchanged
elseif (CMAKE_SIZEOF_VOID_P EQUAL 4)
  if (WAMR_BUILD_TARGET STREQUAL "X86_32")
      add handing if needed
  endif()
else ()
   message(SEND_ERROR "Unsupported build target platform!")
endif ()
```

Reply:
Thanks for the hint, I updated the latest commit of mine after your suggestion. Could you please check & verify that?